hrm-service/src/case/case-sections-sql.ts

```diff
@@ -0,0 +1,42 @@
+import { pgp } from '../connection-pool';
+// eslint-disable-next-line prettier/prettier
+import { CaseSectionRecord } from './case-data-access';
+
+export const SELECT_CASE_SECTIONS = `SELECT 
+         COALESCE(jsonb_agg(DISTINCT cs.*) FILTER (WHERE cs."caseId" IS NOT NULL), '[]') AS "caseSections"
```

Comment:
Possibly, to be honest I migrated the logic from the previous search query and tried changing as little of it as possible, to reduce the result of introducing subtle bugs. There was a similar filter clause in the previous query, which also looked like it might possibly be redundant. Alternatively this type of filter clause may have been required in the previous logic & not be required any more.

It's almost certain that the query can be further refined, but I'd prefer to do this on an iterative basis with a known good starting point, rather than trying to change more than necessary to start with.

Reply:
> but I'd prefer to do this on an iterative basis with a known good starting point

Agreed! All my questions in this PRs (around the migrations) are for learning purposes, not asking any change!